python-checks/src/test/resources/checks/identityComparisonWithCachedTypes.py

```diff
@@ -0,0 +1,153 @@
+    #     ^^
+    param is frozenset([42, 58]) # Noncompliant {{Replace this "is" operator with "=="; identity operator is not reliable here.}}
+    #     ^^
+    frozenset(param) is param # Noncompliant {{Replace this "is" operator with "=="; identity operator is not reliable here.}}
+    #                ^^
+    param is (1, 2, 3) # Noncompliant {{Replace this "is" operator with "=="; identity operator is not reliable here.}}
+    #     ^^
+    (4, 5) is param # Noncompliant {{Replace this "is" operator with "=="; identity operator is not reliable here.}}
+    #      ^^
+
+def literal_comparison_compliant(param):
+    param is param # ok from the point of this rule
+    param is someUnknownFunction(42) # ok
+    param == param # ok
+    param == 2000 # ok
+    0.1 == param # ok
+    param == b'01'
+    "str" == param
+    (4, 5) == param
+
+def literal_comparison_with_not_token(param):
+    param is not 2000  # Noncompliant {{Replace this "is not" operator with "!="; identity operator is not reliable here.}}
+    #     ^^^^^^
+    3000 is not param # Noncompliant {{Replace this "is not" operator with "!="; identity operator is not reliable here.}}
+    #    ^^^^^^
+    param is not (1, 2, 3) # Noncompliant {{Replace this "is not" operator with "!="; identity operator is not reliable here.}}
+    #     ^^^^^^
+
+def literal_comparison_inside_more_interesting_expressions(param):
+  x = 42 if param is not (2, 3) else 58 # Noncompliant {{Replace this "is not" operator with "!="; identity operator is not reliable here.}}
+  #               ^^^^^^
+  y = [x * x for x in range(10) if x is not 5] # Noncompliant {{Replace this "is not" operator with "!="; identity operator is not reliable here.}}
+  #                                  ^^^^^^
+
+def reference_escape_two_method_variables(x, y):
+  x is y # ok, both variables could have come from anywhere, they could actually be identical
+
+def integer_assignment():
+  """This is actually 'true', but it wouldn't hurt to use == here either."""
+  # Even though it adheres to the spec,
+  # this specific test was not in the specification, it could be argued that it's actually valid, just very uncommon.
+  x = 42
+  y = x
+  x is y # Noncompliant {{Replace this "is" operator with "=="; identity operator is not reliable here.}}
+  # ^^
+
+def reference_escape_assignment_through_unknown_function(f):
+  x = 42
+  y = f(x)
+  x is y # ok, `f` could be identity, `x` could have ended up in `y`. It wouldn't be too bad if it was marked, though.
+
+def reference_escape_assignment_into_array(self):
+    a = []
+    x = (1, 2, 3, 4, 5)
+    a[0] = x
+    assert_equal(a[0], x)
+    assert_(a[0] is x)
+
+# Corner cases for code coverage
+def coverage():
+  ambiguous = "" if 42 * 42 < 1700 else (lambda x: x * x)
+  ambiguous is unknown
+  f(x) is g(y)
+  f.g(h) is c.d(e)
+  f is g
+
+
+symbollessName is somethingElse # for `null`-Symbols on names
+
+
+# rest is mutably borrowed from `expected-issues/python/src/RSPEC_5795`.
```

Comment:
I think most of those tests here below are duplicated (for example line 90 we're testing the same as line 2, line 14 and line 96 tests the same thing).
I believe it's important to keep only relevant unit tests for maintenability (less is more!).

Reply:
Made sure that `sort ... | uniq` doesn't show any duplicates. Removed distinction between string literals in single and double quotes.